// @standard-crypto/farcaster-js-hub-rest/dist/esm/openapi/generated/apis/info-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub REST API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Farcaster docs](https://www.thehubble.xyz/docs/httpapi/httpapi.html) for more details. Some client libraries:   - [TypeScript](https://www.npmjs.com/package/@standard-crypto/farcaster-js-hub-rest)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import globalAxios from 'axios';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setSearchParams, toPathString, createRequestFunction } from '../common.js';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base.js';
/**
 * InfoApi - axios parameter creator
 * @export
 */
export const InfoApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Sync Methods
         * @param {boolean} dbstats Whether to return DB stats
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getInfo: async (dbstats, options = {}) => {
            // verify required parameter 'dbstats' is not null or undefined
            assertParamExists('getInfo', 'dbstats', dbstats);
            const localVarPath = `/v1/info`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (dbstats !== undefined) {
                localVarQueryParameter['dbstats'] = dbstats;
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
/**
 * InfoApi - functional programming interface
 * @export
 */
export const InfoApiFp = function (configuration) {
    const localVarAxiosParamCreator = InfoApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @summary Sync Methods
         * @param {boolean} dbstats Whether to return DB stats
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getInfo(dbstats, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getInfo(dbstats, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
    };
};
/**
 * InfoApi - factory interface
 * @export
 */
export const InfoApiFactory = function (configuration, basePath, axios) {
    const localVarFp = InfoApiFp(configuration);
    return {
        /**
         *
         * @summary Sync Methods
         * @param {InfoApiGetInfoRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getInfo(requestParameters, options) {
            return localVarFp.getInfo(requestParameters.dbstats, options).then((request) => request(axios, basePath));
        },
    };
};
/**
 * InfoApi - object-oriented interface
 * @export
 * @class InfoApi
 * @extends {BaseAPI}
 */
export class InfoApi extends BaseAPI {
    /**
     *
     * @summary Sync Methods
     * @param {InfoApiGetInfoRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof InfoApi
     */
    getInfo(requestParameters, options) {
        return InfoApiFp(this.configuration).getInfo(requestParameters.dbstats, options).then((request) => request(this.axios, this.basePath));
    }
}
