// @standard-crypto/farcaster-js-hub-rest/dist/commonjs/openapi/generated/models/signer-event-type.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub REST API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Farcaster docs](https://www.thehubble.xyz/docs/httpapi/httpapi.html) for more details. Some client libraries:   - [TypeScript](https://www.npmjs.com/package/@standard-crypto/farcaster-js-hub-rest)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignerEventType = void 0;
/**
 *
 * @export
 * @enum {string}
 */
var SignerEventType;
(function (SignerEventType) {
    SignerEventType["Add"] = "SIGNER_EVENT_TYPE_ADD";
    SignerEventType["Remove"] = "SIGNER_EVENT_TYPE_REMOVE";
    SignerEventType["AdminReset"] = "SIGNER_EVENT_TYPE_ADMIN_RESET";
})(SignerEventType || (exports.SignerEventType = SignerEventType = {}));
