// @standard-crypto/farcaster-js-hub-rest/dist/commonjs/openapi/generated/models/store-type.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub REST API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Farcaster docs](https://www.thehubble.xyz/docs/httpapi/httpapi.html) for more details. Some client libraries:   - [TypeScript](https://www.npmjs.com/package/@standard-crypto/farcaster-js-hub-rest)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoreType = void 0;
/**
 *
 * @export
 * @enum {string}
 */
var StoreType;
(function (StoreType) {
    StoreType["Casts"] = "STORE_TYPE_CASTS";
    StoreType["Links"] = "STORE_TYPE_LINKS";
    StoreType["Reactions"] = "STORE_TYPE_REACTIONS";
    StoreType["UserData"] = "STORE_TYPE_USER_DATA";
    StoreType["Verifications"] = "STORE_TYPE_VERIFICATIONS";
    StoreType["UsernameProofs"] = "STORE_TYPE_USERNAME_PROOFS";
})(StoreType || (exports.StoreType = StoreType = {}));
