// @standard-crypto/farcaster-js-hub-rest/dist/commonjs/openapi/generated/apis/casts-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub REST API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Farcaster docs](https://www.thehubble.xyz/docs/httpapi/httpapi.html) for more details. Some client libraries:   - [TypeScript](https://www.npmjs.com/package/@standard-crypto/farcaster-js-hub-rest)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CastsApi = exports.CastsApiFactory = exports.CastsApiFp = exports.CastsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_js_1 = require("../common.js");
// @ts-ignore
const base_js_1 = require("../base.js");
/**
 * CastsApi - axios parameter creator
 * @export
 */
const CastsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Get a cast by its FID and Hash.
         * @param {number} fid The FID of the cast\&#39;s creator
         * @param {string} hash The cast\&#39;s hash
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getCastById: async (fid, hash, options = {}) => {
            // verify required parameter 'fid' is not null or undefined
            (0, common_js_1.assertParamExists)('getCastById', 'fid', fid);
            // verify required parameter 'hash' is not null or undefined
            (0, common_js_1.assertParamExists)('getCastById', 'hash', hash);
            const localVarPath = `/v1/castById`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_js_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (hash !== undefined) {
                localVarQueryParameter['hash'] = hash;
            }
            (0, common_js_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_js_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Fetch all casts authored by an FID.
         * @param {number} fid The FID of the casts\&#39; creator
         * @param {number} [pageSize] Maximum number of messages to return in a single response
         * @param {boolean} [reverse] Reverse the sort order, returning latest messages first
         * @param {string} [pageToken] The page token returned by the previous query, to fetch the next page. If this parameter is empty, fetch the first page
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listCastsByFid: async (fid, pageSize, reverse, pageToken, options = {}) => {
            // verify required parameter 'fid' is not null or undefined
            (0, common_js_1.assertParamExists)('listCastsByFid', 'fid', fid);
            const localVarPath = `/v1/castsByFid`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_js_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (pageSize !== undefined) {
                localVarQueryParameter['pageSize'] = pageSize;
            }
            if (reverse !== undefined) {
                localVarQueryParameter['reverse'] = reverse;
            }
            if (pageToken !== undefined) {
                localVarQueryParameter['pageToken'] = pageToken;
            }
            (0, common_js_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_js_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Fetch all casts that mention an FID
         * @param {number} fid The FID that is mentioned in a cast
         * @param {number} [pageSize] Maximum number of messages to return in a single response
         * @param {boolean} [reverse] Reverse the sort order, returning latest messages first
         * @param {string} [pageToken] The page token returned by the previous query, to fetch the next page. If this parameter is empty, fetch the first page
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listCastsByMention: async (fid, pageSize, reverse, pageToken, options = {}) => {
            // verify required parameter 'fid' is not null or undefined
            (0, common_js_1.assertParamExists)('listCastsByMention', 'fid', fid);
            const localVarPath = `/v1/castsByMention`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_js_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (pageSize !== undefined) {
                localVarQueryParameter['pageSize'] = pageSize;
            }
            if (reverse !== undefined) {
                localVarQueryParameter['reverse'] = reverse;
            }
            if (pageToken !== undefined) {
                localVarQueryParameter['pageToken'] = pageToken;
            }
            (0, common_js_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_js_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Fetch all casts by parent cast\'s FID and Hash OR by the parent\'s URL
         * @param {number} [fid] The FID of the parent cast
         * @param {string} [hash] The parent cast\&#39;s hash
         * @param {string} [url]
         * @param {number} [pageSize] Maximum number of messages to return in a single response
         * @param {boolean} [reverse] Reverse the sort order, returning latest messages first
         * @param {string} [pageToken] The page token returned by the previous query, to fetch the next page. If this parameter is empty, fetch the first page
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listCastsByParent: async (fid, hash, url, pageSize, reverse, pageToken, options = {}) => {
            const localVarPath = `/v1/castsByParent`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_js_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (hash !== undefined) {
                localVarQueryParameter['hash'] = hash;
            }
            if (url !== undefined) {
                localVarQueryParameter['url'] = url;
            }
            if (pageSize !== undefined) {
                localVarQueryParameter['pageSize'] = pageSize;
            }
            if (reverse !== undefined) {
                localVarQueryParameter['reverse'] = reverse;
            }
            if (pageToken !== undefined) {
                localVarQueryParameter['pageToken'] = pageToken;
            }
            (0, common_js_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_js_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.CastsApiAxiosParamCreator = CastsApiAxiosParamCreator;
/**
 * CastsApi - functional programming interface
 * @export
 */
const CastsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.CastsApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Get a cast by its FID and Hash.
         * @param {number} fid The FID of the cast\&#39;s creator
         * @param {string} hash The cast\&#39;s hash
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getCastById(fid, hash, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getCastById(fid, hash, options);
            return (0, common_js_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_js_1.BASE_PATH, configuration);
        },
        /**
         *
         * @summary Fetch all casts authored by an FID.
         * @param {number} fid The FID of the casts\&#39; creator
         * @param {number} [pageSize] Maximum number of messages to return in a single response
         * @param {boolean} [reverse] Reverse the sort order, returning latest messages first
         * @param {string} [pageToken] The page token returned by the previous query, to fetch the next page. If this parameter is empty, fetch the first page
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listCastsByFid(fid, pageSize, reverse, pageToken, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listCastsByFid(fid, pageSize, reverse, pageToken, options);
            return (0, common_js_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_js_1.BASE_PATH, configuration);
        },
        /**
         *
         * @summary Fetch all casts that mention an FID
         * @param {number} fid The FID that is mentioned in a cast
         * @param {number} [pageSize] Maximum number of messages to return in a single response
         * @param {boolean} [reverse] Reverse the sort order, returning latest messages first
         * @param {string} [pageToken] The page token returned by the previous query, to fetch the next page. If this parameter is empty, fetch the first page
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listCastsByMention(fid, pageSize, reverse, pageToken, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listCastsByMention(fid, pageSize, reverse, pageToken, options);
            return (0, common_js_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_js_1.BASE_PATH, configuration);
        },
        /**
         *
         * @summary Fetch all casts by parent cast\'s FID and Hash OR by the parent\'s URL
         * @param {number} [fid] The FID of the parent cast
         * @param {string} [hash] The parent cast\&#39;s hash
         * @param {string} [url]
         * @param {number} [pageSize] Maximum number of messages to return in a single response
         * @param {boolean} [reverse] Reverse the sort order, returning latest messages first
         * @param {string} [pageToken] The page token returned by the previous query, to fetch the next page. If this parameter is empty, fetch the first page
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listCastsByParent(fid, hash, url, pageSize, reverse, pageToken, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listCastsByParent(fid, hash, url, pageSize, reverse, pageToken, options);
            return (0, common_js_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_js_1.BASE_PATH, configuration);
        },
    };
};
exports.CastsApiFp = CastsApiFp;
/**
 * CastsApi - factory interface
 * @export
 */
const CastsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.CastsApiFp)(configuration);
    return {
        /**
         *
         * @summary Get a cast by its FID and Hash.
         * @param {CastsApiGetCastByIdRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getCastById(requestParameters, options) {
            return localVarFp.getCastById(requestParameters.fid, requestParameters.hash, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Fetch all casts authored by an FID.
         * @param {CastsApiListCastsByFidRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listCastsByFid(requestParameters, options) {
            return localVarFp.listCastsByFid(requestParameters.fid, requestParameters.pageSize, requestParameters.reverse, requestParameters.pageToken, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Fetch all casts that mention an FID
         * @param {CastsApiListCastsByMentionRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listCastsByMention(requestParameters, options) {
            return localVarFp.listCastsByMention(requestParameters.fid, requestParameters.pageSize, requestParameters.reverse, requestParameters.pageToken, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Fetch all casts by parent cast\'s FID and Hash OR by the parent\'s URL
         * @param {CastsApiListCastsByParentRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listCastsByParent(requestParameters = {}, options) {
            return localVarFp.listCastsByParent(requestParameters.fid, requestParameters.hash, requestParameters.url, requestParameters.pageSize, requestParameters.reverse, requestParameters.pageToken, options).then((request) => request(axios, basePath));
        },
    };
};
exports.CastsApiFactory = CastsApiFactory;
/**
 * CastsApi - object-oriented interface
 * @export
 * @class CastsApi
 * @extends {BaseAPI}
 */
class CastsApi extends base_js_1.BaseAPI {
    /**
     *
     * @summary Get a cast by its FID and Hash.
     * @param {CastsApiGetCastByIdRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof CastsApi
     */
    getCastById(requestParameters, options) {
        return (0, exports.CastsApiFp)(this.configuration).getCastById(requestParameters.fid, requestParameters.hash, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Fetch all casts authored by an FID.
     * @param {CastsApiListCastsByFidRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof CastsApi
     */
    listCastsByFid(requestParameters, options) {
        return (0, exports.CastsApiFp)(this.configuration).listCastsByFid(requestParameters.fid, requestParameters.pageSize, requestParameters.reverse, requestParameters.pageToken, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Fetch all casts that mention an FID
     * @param {CastsApiListCastsByMentionRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof CastsApi
     */
    listCastsByMention(requestParameters, options) {
        return (0, exports.CastsApiFp)(this.configuration).listCastsByMention(requestParameters.fid, requestParameters.pageSize, requestParameters.reverse, requestParameters.pageToken, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Fetch all casts by parent cast\'s FID and Hash OR by the parent\'s URL
     * @param {CastsApiListCastsByParentRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof CastsApi
     */
    listCastsByParent(requestParameters = {}, options) {
        return (0, exports.CastsApiFp)(this.configuration).listCastsByParent(requestParameters.fid, requestParameters.hash, requestParameters.url, requestParameters.pageSize, requestParameters.reverse, requestParameters.pageToken, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.CastsApi = CastsApi;
