// @standard-crypto/farcaster-js-hub-rest/dist/commonjs/openapi/generated/apis/fids-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub REST API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Farcaster docs](https://www.thehubble.xyz/docs/httpapi/httpapi.html) for more details. Some client libraries:   - [TypeScript](https://www.npmjs.com/package/@standard-crypto/farcaster-js-hub-rest)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FIDsApi = exports.FIDsApiFactory = exports.FIDsApiFp = exports.FIDsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_js_1 = require("../common.js");
// @ts-ignore
const base_js_1 = require("../base.js");
/**
 * FIDsApi - axios parameter creator
 * @export
 */
const FIDsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Get a list of all the FIDs
         * @param {number} [pageSize] Maximum number of messages to return in a single response
         * @param {boolean} [reverse] Reverse the sort order, returning latest messages first
         * @param {string} [pageToken] The page token returned by the previous query, to fetch the next page. If this parameter is empty, fetch the first page
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listFids: async (pageSize, reverse, pageToken, options = {}) => {
            const localVarPath = `/v1/fids`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_js_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (pageSize !== undefined) {
                localVarQueryParameter['pageSize'] = pageSize;
            }
            if (reverse !== undefined) {
                localVarQueryParameter['reverse'] = reverse;
            }
            if (pageToken !== undefined) {
                localVarQueryParameter['pageToken'] = pageToken;
            }
            (0, common_js_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_js_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.FIDsApiAxiosParamCreator = FIDsApiAxiosParamCreator;
/**
 * FIDsApi - functional programming interface
 * @export
 */
const FIDsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.FIDsApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Get a list of all the FIDs
         * @param {number} [pageSize] Maximum number of messages to return in a single response
         * @param {boolean} [reverse] Reverse the sort order, returning latest messages first
         * @param {string} [pageToken] The page token returned by the previous query, to fetch the next page. If this parameter is empty, fetch the first page
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listFids(pageSize, reverse, pageToken, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listFids(pageSize, reverse, pageToken, options);
            return (0, common_js_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_js_1.BASE_PATH, configuration);
        },
    };
};
exports.FIDsApiFp = FIDsApiFp;
/**
 * FIDsApi - factory interface
 * @export
 */
const FIDsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.FIDsApiFp)(configuration);
    return {
        /**
         *
         * @summary Get a list of all the FIDs
         * @param {FIDsApiListFidsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listFids(requestParameters = {}, options) {
            return localVarFp.listFids(requestParameters.pageSize, requestParameters.reverse, requestParameters.pageToken, options).then((request) => request(axios, basePath));
        },
    };
};
exports.FIDsApiFactory = FIDsApiFactory;
/**
 * FIDsApi - object-oriented interface
 * @export
 * @class FIDsApi
 * @extends {BaseAPI}
 */
class FIDsApi extends base_js_1.BaseAPI {
    /**
     *
     * @summary Get a list of all the FIDs
     * @param {FIDsApiListFidsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FIDsApi
     */
    listFids(requestParameters = {}, options) {
        return (0, exports.FIDsApiFp)(this.configuration).listFids(requestParameters.pageSize, requestParameters.reverse, requestParameters.pageToken, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.FIDsApi = FIDsApi;
