// @standard-crypto/farcaster-js-hub-rest/dist/commonjs/openapi/generated/apis/info-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub REST API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Farcaster docs](https://www.thehubble.xyz/docs/httpapi/httpapi.html) for more details. Some client libraries:   - [TypeScript](https://www.npmjs.com/package/@standard-crypto/farcaster-js-hub-rest)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InfoApi = exports.InfoApiFactory = exports.InfoApiFp = exports.InfoApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_js_1 = require("../common.js");
// @ts-ignore
const base_js_1 = require("../base.js");
/**
 * InfoApi - axios parameter creator
 * @export
 */
const InfoApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Sync Methods
         * @param {boolean} dbstats Whether to return DB stats
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getInfo: async (dbstats, options = {}) => {
            // verify required parameter 'dbstats' is not null or undefined
            (0, common_js_1.assertParamExists)('getInfo', 'dbstats', dbstats);
            const localVarPath = `/v1/info`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_js_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (dbstats !== undefined) {
                localVarQueryParameter['dbstats'] = dbstats;
            }
            (0, common_js_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_js_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.InfoApiAxiosParamCreator = InfoApiAxiosParamCreator;
/**
 * InfoApi - functional programming interface
 * @export
 */
const InfoApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.InfoApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Sync Methods
         * @param {boolean} dbstats Whether to return DB stats
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getInfo(dbstats, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getInfo(dbstats, options);
            return (0, common_js_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_js_1.BASE_PATH, configuration);
        },
    };
};
exports.InfoApiFp = InfoApiFp;
/**
 * InfoApi - factory interface
 * @export
 */
const InfoApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.InfoApiFp)(configuration);
    return {
        /**
         *
         * @summary Sync Methods
         * @param {InfoApiGetInfoRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getInfo(requestParameters, options) {
            return localVarFp.getInfo(requestParameters.dbstats, options).then((request) => request(axios, basePath));
        },
    };
};
exports.InfoApiFactory = InfoApiFactory;
/**
 * InfoApi - object-oriented interface
 * @export
 * @class InfoApi
 * @extends {BaseAPI}
 */
class InfoApi extends base_js_1.BaseAPI {
    /**
     *
     * @summary Sync Methods
     * @param {InfoApiGetInfoRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof InfoApi
     */
    getInfo(requestParameters, options) {
        return (0, exports.InfoApiFp)(this.configuration).getInfo(requestParameters.dbstats, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.InfoApi = InfoApi;
