// @standard-crypto/farcaster-js-hub-rest/dist/commonjs/openapi/generated/apis/storage-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub REST API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Farcaster docs](https://www.thehubble.xyz/docs/httpapi/httpapi.html) for more details. Some client libraries:   - [TypeScript](https://www.npmjs.com/package/@standard-crypto/farcaster-js-hub-rest)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageApi = exports.StorageApiFactory = exports.StorageApiFp = exports.StorageApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_js_1 = require("../common.js");
// @ts-ignore
const base_js_1 = require("../base.js");
/**
 * StorageApi - axios parameter creator
 * @export
 */
const StorageApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Get an FID\'s storage limits.
         * @param {number} fid
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getStorageLimitsByFid: async (fid, options = {}) => {
            // verify required parameter 'fid' is not null or undefined
            (0, common_js_1.assertParamExists)('getStorageLimitsByFid', 'fid', fid);
            const localVarPath = `/v1/storageLimitsByFid`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_js_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            (0, common_js_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_js_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.StorageApiAxiosParamCreator = StorageApiAxiosParamCreator;
/**
 * StorageApi - functional programming interface
 * @export
 */
const StorageApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.StorageApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Get an FID\'s storage limits.
         * @param {number} fid
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getStorageLimitsByFid(fid, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getStorageLimitsByFid(fid, options);
            return (0, common_js_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_js_1.BASE_PATH, configuration);
        },
    };
};
exports.StorageApiFp = StorageApiFp;
/**
 * StorageApi - factory interface
 * @export
 */
const StorageApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.StorageApiFp)(configuration);
    return {
        /**
         *
         * @summary Get an FID\'s storage limits.
         * @param {StorageApiGetStorageLimitsByFidRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getStorageLimitsByFid(requestParameters, options) {
            return localVarFp.getStorageLimitsByFid(requestParameters.fid, options).then((request) => request(axios, basePath));
        },
    };
};
exports.StorageApiFactory = StorageApiFactory;
/**
 * StorageApi - object-oriented interface
 * @export
 * @class StorageApi
 * @extends {BaseAPI}
 */
class StorageApi extends base_js_1.BaseAPI {
    /**
     *
     * @summary Get an FID\'s storage limits.
     * @param {StorageApiGetStorageLimitsByFidRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StorageApi
     */
    getStorageLimitsByFid(requestParameters, options) {
        return (0, exports.StorageApiFp)(this.configuration).getStorageLimitsByFid(requestParameters.fid, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.StorageApi = StorageApi;
