// @standard-crypto/farcaster-js-hub-rest/dist/esm/openapi/generated/models/farcaster-network.js
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub REST API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Farcaster docs](https://www.thehubble.xyz/docs/httpapi/httpapi.html) for more details. Some client libraries:   - [TypeScript](https://www.npmjs.com/package/@standard-crypto/farcaster-js-hub-rest)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Farcaster network the message is intended for. - FARCASTER_NETWORK_MAINNET: Public primary network  - FARCASTER_NETWORK_TESTNET: Public test network  - FARCASTER_NETWORK_DEVNET: Private test network
 * @export
 * @enum {string}
 */
export var FarcasterNetwork;
(function (FarcasterNetwork) {
    FarcasterNetwork["Mainnet"] = "FARCASTER_NETWORK_MAINNET";
    FarcasterNetwork["Testnet"] = "FARCASTER_NETWORK_TESTNET";
    FarcasterNetwork["Devnet"] = "FARCASTER_NETWORK_DEVNET";
})(FarcasterNetwork || (FarcasterNetwork = {}));
