// @standard-crypto/farcaster-js-hub-rest/dist/esm/openapi/generated/models/message-type.js
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub REST API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Farcaster docs](https://www.thehubble.xyz/docs/httpapi/httpapi.html) for more details. Some client libraries:   - [TypeScript](https://www.npmjs.com/package/@standard-crypto/farcaster-js-hub-rest)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Type of the MessageBody.  - MESSAGE_TYPE_CAST_ADD: Add a new Cast  - MESSAGE_TYPE_CAST_REMOVE: Remove an existing Cast  - MESSAGE_TYPE_REACTION_ADD: Add a Reaction to a Cast  - MESSAGE_TYPE_REACTION_REMOVE: Remove a Reaction from a Cast  - MESSAGE_TYPE_LINK_ADD: Add a new Link  - MESSAGE_TYPE_LINK_REMOVE: Remove an existing Link  - MESSAGE_TYPE_VERIFICATION_ADD_ETH_ADDRESS: Add a Verification of an Ethereum Address  - MESSAGE_TYPE_VERIFICATION_REMOVE: Remove a Verification  - MESSAGE_TYPE_USER_DATA_ADD: Add metadata about a user  - MESSAGE_TYPE_USERNAME_PROOF: Add or replace a username proof  - MESSAGE_TYPE_FRAME_ACTION: A Farcaster Frame action
 * @export
 * @enum {string}
 */
export var MessageType;
(function (MessageType) {
    MessageType["CastAdd"] = "MESSAGE_TYPE_CAST_ADD";
    MessageType["CastRemove"] = "MESSAGE_TYPE_CAST_REMOVE";
    MessageType["ReactionAdd"] = "MESSAGE_TYPE_REACTION_ADD";
    MessageType["ReactionRemove"] = "MESSAGE_TYPE_REACTION_REMOVE";
    MessageType["LinkAdd"] = "MESSAGE_TYPE_LINK_ADD";
    MessageType["LinkRemove"] = "MESSAGE_TYPE_LINK_REMOVE";
    MessageType["VerificationAddEthAddress"] = "MESSAGE_TYPE_VERIFICATION_ADD_ETH_ADDRESS";
    MessageType["VerificationRemove"] = "MESSAGE_TYPE_VERIFICATION_REMOVE";
    MessageType["UserDataAdd"] = "MESSAGE_TYPE_USER_DATA_ADD";
    MessageType["UsernameProof"] = "MESSAGE_TYPE_USERNAME_PROOF";
    MessageType["FrameAction"] = "MESSAGE_TYPE_FRAME_ACTION";
})(MessageType || (MessageType = {}));
