// @standard-crypto/farcaster-js-hub-rest/dist/esm/openapi/generated/models/user-data-type.js
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub REST API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Farcaster docs](https://www.thehubble.xyz/docs/httpapi/httpapi.html) for more details. Some client libraries:   - [TypeScript](https://www.npmjs.com/package/@standard-crypto/farcaster-js-hub-rest)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Type of UserData. - USER_DATA_TYPE_PFP: Profile Picture for the user  - USER_DATA_TYPE_DISPLAY: Display Name for the user  - USER_DATA_TYPE_BIO: Bio for the user  - USER_DATA_TYPE_URL: URL of the user  - USER_DATA_TYPE_USERNAME: Preferred Name for the user
 * @export
 * @enum {string}
 */
export var UserDataType;
(function (UserDataType) {
    UserDataType["Pfp"] = "USER_DATA_TYPE_PFP";
    UserDataType["Display"] = "USER_DATA_TYPE_DISPLAY";
    UserDataType["Bio"] = "USER_DATA_TYPE_BIO";
    UserDataType["Url"] = "USER_DATA_TYPE_URL";
    UserDataType["Username"] = "USER_DATA_TYPE_USERNAME";
})(UserDataType || (UserDataType = {}));
