// @standard-crypto/farcaster-js-neynar/dist/commonjs/v1/openapi/generated/apis/verification-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * v1 Farcaster
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerificationApi = exports.VerificationApiFactory = exports.VerificationApiFp = exports.VerificationApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_js_1 = require("../common.js");
// @ts-ignore
const base_js_1 = require("../base.js");
/**
 * VerificationApi - axios parameter creator
 * @export
 */
const VerificationApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Checks if a given Ethereum address has a Farcaster user associated with it. Note: if an address is associated with multiple users, the API will return the user who most recently published a verification with the address (based on when Warpcast received the proof, not a self-reported timestamp).
         * @summary Retrieve user for a given ethereum address
         * @param {string} address
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        userByVerification: async (address, options = {}) => {
            // verify required parameter 'address' is not null or undefined
            (0, common_js_1.assertParamExists)('userByVerification', 'address', address);
            const localVarPath = `/farcaster/user-by-verification`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_js_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_js_1.setApiKeyToObject)(localVarHeaderParameter, "api_key", configuration);
            if (address !== undefined) {
                localVarQueryParameter['address'] = address;
            }
            (0, common_js_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_js_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get all known verifications of a user
         * @summary Retrieve verifications for a given FID
         * @param {number} fid FID of the user
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        verifications: async (fid, options = {}) => {
            // verify required parameter 'fid' is not null or undefined
            (0, common_js_1.assertParamExists)('verifications', 'fid', fid);
            const localVarPath = `/farcaster/verifications`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_js_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_js_1.setApiKeyToObject)(localVarHeaderParameter, "api_key", configuration);
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            (0, common_js_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_js_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.VerificationApiAxiosParamCreator = VerificationApiAxiosParamCreator;
/**
 * VerificationApi - functional programming interface
 * @export
 */
const VerificationApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.VerificationApiAxiosParamCreator)(configuration);
    return {
        /**
         * Checks if a given Ethereum address has a Farcaster user associated with it. Note: if an address is associated with multiple users, the API will return the user who most recently published a verification with the address (based on when Warpcast received the proof, not a self-reported timestamp).
         * @summary Retrieve user for a given ethereum address
         * @param {string} address
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async userByVerification(address, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.userByVerification(address, options);
            return (0, common_js_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_js_1.BASE_PATH, configuration);
        },
        /**
         * Get all known verifications of a user
         * @summary Retrieve verifications for a given FID
         * @param {number} fid FID of the user
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async verifications(fid, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.verifications(fid, options);
            return (0, common_js_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_js_1.BASE_PATH, configuration);
        },
    };
};
exports.VerificationApiFp = VerificationApiFp;
/**
 * VerificationApi - factory interface
 * @export
 */
const VerificationApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.VerificationApiFp)(configuration);
    return {
        /**
         * Checks if a given Ethereum address has a Farcaster user associated with it. Note: if an address is associated with multiple users, the API will return the user who most recently published a verification with the address (based on when Warpcast received the proof, not a self-reported timestamp).
         * @summary Retrieve user for a given ethereum address
         * @param {VerificationApiUserByVerificationRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        userByVerification(requestParameters, options) {
            return localVarFp.userByVerification(requestParameters.address, options).then((request) => request(axios, basePath));
        },
        /**
         * Get all known verifications of a user
         * @summary Retrieve verifications for a given FID
         * @param {VerificationApiVerificationsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        verifications(requestParameters, options) {
            return localVarFp.verifications(requestParameters.fid, options).then((request) => request(axios, basePath));
        },
    };
};
exports.VerificationApiFactory = VerificationApiFactory;
/**
 * VerificationApi - object-oriented interface
 * @export
 * @class VerificationApi
 * @extends {BaseAPI}
 */
class VerificationApi extends base_js_1.BaseAPI {
    /**
     * Checks if a given Ethereum address has a Farcaster user associated with it. Note: if an address is associated with multiple users, the API will return the user who most recently published a verification with the address (based on when Warpcast received the proof, not a self-reported timestamp).
     * @summary Retrieve user for a given ethereum address
     * @param {VerificationApiUserByVerificationRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof VerificationApi
     */
    userByVerification(requestParameters, options) {
        return (0, exports.VerificationApiFp)(this.configuration).userByVerification(requestParameters.address, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get all known verifications of a user
     * @summary Retrieve verifications for a given FID
     * @param {VerificationApiVerificationsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof VerificationApi
     */
    verifications(requestParameters, options) {
        return (0, exports.VerificationApiFp)(this.configuration).verifications(requestParameters.fid, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.VerificationApi = VerificationApi;
